holopy/inference/interface.py

```diff
@@ -22,28 +22,57 @@
 
 from holopy.scattering import Scatterer
 from holopy.scattering.scatterer import _interpret_parameters
-from holopy.inference.model import ALL_STRATEGIES, Model, AlphaModel
+from holopy.inference.model import Model, AlphaModel
 from holopy.inference.prior import Uniform
+from holopy.inference.nmpfit import NmpfitStrategy
+from holopy.inference.scipyfit import LeastSquaresScipyStrategy
+from holopy.inference.cmaes import CmaStrategy
+from holopy.inference.emcee import EmceeStrategy, TemperedStrategy
+
+DEFAULT_STRATEGY = {'fit': 'nmpfit', 'sample': 'emcee'}
+ALL_STRATEGIES = {'fit': {'nmpfit': NmpfitStrategy,
+                          'scipy lsq': LeastSquaresScipyStrategy,
+                          'cma': CmaStrategy},
+                  'sample': {'emcee': EmceeStrategy,
+                             'subset tempering': TemperedStrategy,
+                             'parallel tempering': NotImplemented}}
 
 available_fit_strategies = ALL_STRATEGIES['fit']
 available_sampling_strategies = ALL_STRATEGIES['sample']
 
 
-def sample(data, model):
+def sample(data, model, strategy=None):
     if isinstance(model, Model):
-        return model.sample(data)
+        strategy = validate_strategy(strategy, 'sample')
+        return strategy.sample(model, data)
     else:
         msg = "Sampling model {} is not a HoloPy Model object.".format(model)
         raise ValueError(msg)
 
 
-def fit(data, model, parameters=None):
+def fit(data, model, parameters=None, strategy=None):
     if isinstance(model, Scatterer):
         model = make_default_model(model, parameters)
     elif parameters is not None:
         warnings.warn("Ignoring parameters {} in favour of model {}.".format(
                       parameters, model), UserWarning)
-    return model.fit(data)
+    strategy = validate_strategy(strategy, 'fit')
+    return strategy.fit(model, data)
+
+
+def validate_strategy(strategy, operation):
+    if strategy is None:
+        strategy = DEFAULT_STRATEGY[operation]
+    if isinstance(strategy, str):
+        strategy = ALL_STRATEGIES[operation][strategy]
+    if not hasattr(strategy, operation):
+        raise ValueError("Cannot {} with Strategy of type {}.".format(
+            operation, type(strategy).__name__))
+    try:
+        strategy = strategy()
+    except TypeError:
```

Comment:
I'm a little worried by this, because there could conceivably be a TypeEror in the strategy. I think an `if isinstance()` call might be better here.